### PR TITLE
Addition of templates for Cisco s300 switches.

### PR DESCRIPTION
**Pull request by alphabet5**
Templates created for lldp, int status, and mac address-table for Cisco s300 switches.

##### ISSUE TYPE

 - New Template Pull Request

##### COMPONENT

cisco_s300_show_interfaces_status.template, 'sh int status' on knock-off ios on a Cisco s300.
cisco_s300_show_mac_address_table.template, 'sh mac add' on knock-off ios on a Cisco s300.
cisco_s300_show_lldp_neighbors.template, 'sh lldp neigh' on knock-off ios on a Cisco s300.
##### SUMMARY

No changes, just new templates to use with the differing cli output on s300 switches. 

```

```

### CODE REVIEW

**Comment by FragmentedPacket** [comment]
@alphabet5 Just wanted to check in on this. As Jacob stated, could you add those necessary lines and tests to this PR and then we can work on getting this merged in?

**Comment by alphabet5** [comment]
After adding the 2 lines to the end, I'm getting errors when processing the raw input with textfsm.

.raw first few lines

`                                             Flow Link          Back   Mdix
Port     Type         Duplex  Speed Neg      ctrl State       Pressure Mode
-------- ------------ ------  ----- -------- ---- ----------- -------- -------
gi1      1G-Copper    Full    100   Enabled  Off  Up          Disabled Off
gi2      1G-Copper    Full    100   Enabled  Off  Up          Disabled On
gi3      1G-Copper    Full    100   Enabled  Off  Up          Disabled Off`

textfsm template

`Value PORT (\S+)
Value TYPE (\S+)
Value DUPLEX (Full|Half|--)
Value SPEED (\d+|--)
Value NEG (Enabled|Disabled|--)
Value FLOWCTRL (Off|On|--)
Value LINKSTATE (Up|Down)
Value BACKPRESSURE (Disabled|Enabled|--)
Value MDIXMODE (Off|On|--)

Start
  ^${PORT}\s+${TYPE}\s+${DUPLEX}\s+${SPEED}\s+${NEG}\s+${FLOWCTRL}\s+${LINKSTATE}\s+${BACKPRESSURE}\s+${MDIXMODE} -> Record
  ^\s+$$
  ^. -> Error
`

Python error.

`textfsm.TextFSM(open('templates/cisco_s300_show_interfaces_status.template')).ParseText(open('tests/cisco_s300/show_interfaces_status/cisco_s300_show_interfaces_status.raw', 'r').read())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/textfsm/parser.py", line 895, in ParseText
    self._CheckLine(line)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/textfsm/parser.py", line 944, in _CheckLine
    if self._Operations(rule, line):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/textfsm/parser.py", line 1024, in _Operations
    raise TextFSMError('State Error raised. Rule Line: %s. Input Line: %s'
textfsm.parser.TextFSMError: State Error raised. Rule Line: 14. Input Line:                                              Flow Link          Back   Mdix
`

It's probably going to take a bit to sort through this, as I have other priorities right now. Any guidance would be appreciated. I believe it's due to the header being separated into two rows, I'm just not quite sure how to fix it.

**Comment by FragmentedPacket** [comment]
Can you please include the `.raw` files so we can better assist? Pushing up to PR would be fine.

File: templates/cisco_s300_show_interfaces_status.textfsm [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Value NEG (Enabled|Disabled|--)
+Value FLOWCTRL (Off|On|--)
+Value LINKSTATE (Up|Down)
+Value BACKPRESSURE (Disabled|Enabled|--)
+Value MDIXMODE (Off|On|--)
+
+Start
+  ^\s
+  ^Port -> Begin
+  ^\s+$$
```
Comment (FragmentedPacket) [inline templates/cisco_s300_show_interfaces_status.textfsm:14]:
```suggestion
  ^\s*$$
```

Do you mind updating each template to this?

File: templates/cisco_s300_show_interfaces_status.textfsm [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  ^. -> Error
+
+Begin
+  ^${PORT}\s+${TYPE}\s+${DUPLEX}\s+${SPEED}\s+${NEG}\s+${FLOWCTRL}\s+${LINKSTATE}\s+${BACKPRESSURE}\s+${MDIXMODE} -> Record
```
Comment (FragmentedPacket) [inline templates/cisco_s300_show_interfaces_status.textfsm:18]: Can you add a state transition to end once it hits the table below it?

```
  ^\s*$$
  ^\s+Flow\s+Link -> End
  ^. -> Error
```

File: templates/cisco_s300_show_lldp_neighbors.textfsm [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Value NEIGHBOR (\S+)
+Value LOCAL_INTERFACE (\S+)
+Value NEIGHBOR_INTERFACE (\S+)
+Value DEVICE_ID (([0-9a-f]{2}[:-]){5}([0-9a-f]{2}))
+
+Start
+  ^System capability legend:
```
Comment (FragmentedPacket) [inline templates/cisco_s300_show_lldp_neighbors.textfsm:7]: Do you mind taking out the static spacing and using `\s+` here?

```suggestion
  ^System\s+capability\s+legend:
  ^\S+\s-
  ^\s*Port\s+Device\s+ID\s+Port\s+ID\s+System\s+Name\s+Capabilities\s+TTL -> Begin
  ^\s*$$
  ^. -> Error

Begin
  ^-+
  ^${LOCAL_INTERFACE}\s+${DEVICE_ID}\s+${NEIGHBOR_INTERFACE}\s+${NEIGHBOR}.+ -> Record
  ^\s*$$
  ^. -> Error
```

By changing the state on the header, we can assure the data will be in the necessary format for parsing.

File: templates/cisco_s300_show_mac_address-table.textfsm [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Value DESTINATION_ADDRESS ((\w\w:){5}\w\w)
+Value TYPE (dynamic|self)
+Value VLAN (\w+)
+Value DESTINATION_PORT (\S+)
+
```
Comment (FragmentedPacket) [inline templates/cisco_s300_show_mac_address-table.textfsm:5]:
```suggestion
Start
  ^Flags:
  ^Aging
  ^\s+Vlan\s+Mac\s+Address\s+Port\s+Type -> Begin
  ^\s*$$
  ^. -> Error

Begin
  ^-+
  ^\s+${VLAN}\s+${DESTINATION_ADDRESS}\s+${DESTINATION_PORT}\s+${TYPE} -> Record
  ^\s*$$
  ^. -> Error
```

**Comment by FragmentedPacket** [comment]
Moving work to #719

**Pull request closed without merging**